src/Json.Schema.ToDotNet/ClassGenerator.cs

```diff
@@ -381,6 +402,95 @@ private ConstructorDeclarationSyntax GenerateDefaultConstructor()
+        /// An array containing one initialization statement for each property
+        /// for which the schema specifies a default value.
+        /// </returns>
+        private ExpressionStatementSyntax[] GenerateDefaultInitializations()
+        {
+            var initializations = new List<ExpressionStatementSyntax>();
+
+            foreach (string propertyName in PropInfoDictionary.GetPropertyNames())
+            {
+                if (IncludeProperty(propertyName))
+                {
+                    PropertyInfo propInfo = PropInfoDictionary[propertyName];
+                    object defaultValue = propInfo.DefaultValue;
+
+                    ExpressionStatementSyntax initializationStatement = GenerateDefaultInitialization(propertyName, defaultValue);
+                    if (initializationStatement != null)
+                    {
+                        initializations.Add(GenerateDefaultInitialization(propertyName, defaultValue));
+                    }
+                }
+            }
+
+            return initializations.ToArray();
+        }
+
+        private ExpressionStatementSyntax GenerateDefaultInitialization(
+            string propertyName,
+            object defaultValue)
+        {
+            LiteralExpressionSyntax defaultValueExpression = GetLiteralExpressionForValue(defaultValue);
+            if (defaultValueExpression == null) { return null; }
+
+            return SyntaxFactory.ExpressionStatement(
+                SyntaxFactory.AssignmentExpression(
+                    SyntaxKind.SimpleAssignmentExpression,
+                    SyntaxFactory.IdentifierName(propertyName),
+                    defaultValueExpression));
+        }
+
+        private LiteralExpressionSyntax GetLiteralExpressionForValue(object value)
+        {
+            LiteralExpressionSyntax literalExpression = null;
+
+            if (value is bool)
+            {
+                SyntaxKind literalSyntaxKind = (bool)value == true
+                    ? SyntaxKind.TrueLiteralExpression
+                    : SyntaxKind.FalseLiteralExpression;
+
+                literalExpression = SyntaxFactory.LiteralExpression(literalSyntaxKind);
+            }
+            else if (value is long)
+            {
+                literalExpression = SyntaxFactory.LiteralExpression(
+                    SyntaxKind.NumericLiteralExpression,
+                    SyntaxFactory.Literal((int)(long)value));
+                // Note: the extra cast compensates for a mismatch between our code generation
+                // and Newtonsoft.Json's deserialization behavior. Newtonsoft deserializes
+                // integer properties as Int64 (long), but we generate integer properties
+                // with type int. The extra cast causes Roslyn to emit the literal 42,
+                // which can be assigned to an int, rather than 42L, which cannot. We should
+                // consider changing the code generation to emit longs for integer properties.
```

Comment:
>. [](start = 93, length = 1)

Michael, your thoughts on this?

Reply:
I see. Yes, the problem is ours, JSON/JavaScript number/integer primitives clearly allow for values that exceed C# Int32. If we're going to make this change we should do it soon! It will be an intrusive breaking change. Since we're expanding the type size, the impact should be minimal beyond the literal type change. But it's worrisome to think about the OM signature impact here.

---
In reply to: [244400469](https://github.com/Microsoft/jschema/pull/60#discussion_r244400469) [](ancestors = 244400469)

Reply:
Michael argued that .NET pervasively uses `int` (not `long`, not even `uint)` in its APIs -- and that for historical reasons (because VB didn't support "unsigned"). We need to think hard about this, and assess the impact on interoperability with other .NET APIs, before deciding to do this. Won't do it now.

---
In reply to: [244417379](https://github.com/Microsoft/jschema/pull/60#discussion_r244417379) [](ancestors = 244417379,244400469)